src/cli/commands/force/lightning/local/device/list.ts

```diff
@@ -62,29 +62,25 @@ export default class List extends SfdxCommand {
             );
         }
 
-        return new Promise<any>((resolve, reject) => {
-            const deviceList = CommandLineUtils.platformFlagIsIOS(platform)
-                ? this.iOSDeviceList()
-                : this.androidDeviceList();
-
-            resolve(deviceList);
-        });
+        return CommandLineUtils.platformFlagIsIOS(platform)
```

Comment:
This is another pattern that I'm uncertain about—more specifically, for the entirety of this method.  And it boils down to this: functionally/executionally, what's the difference between this:
```typescript
async function asyncThing(): Promise<boolean> {
    const retVal = false;
    return Promise.resolve(retVal);
}
```
And this:
```typescript
async function asyncThing(): Promise<boolean> {
    return new Promise<boolean>((resolve, reject) => {
        const retVal = false;
        resolve(retVal);
    });
}
```
Because my gut says that these will actually exhibit different behaviors—if subtly—at runtime. [This SO answer](https://stackoverflow.com/a/27716590) seems to run through the behavior differences? But I'm still parsing it to understand better what it means at runtime.

Reply:
Oops, this is more applicable below. I'll move the comment to the right place.